src/chart-review/chart-review.component.tsx

```diff
@@ -1,87 +1,98 @@
-*/
+  const [routes, setRoutes] = React.useState([]);
+  const [navbarItems, setNavbarItems] = React.useState([]);
 
   const [selected, setSelected] = React.useState(getInitialTab());
-
-  React.useEffect(() => {
-    const externalWidgets: externalWidgetsType = {};
-    const promises = [];
-    const moduleMap = {};
-
-    config.widgetDefinitions.forEach(def => {
-      externalWidgets[def.name] = def;
-      //only import modules once
-      if (moduleMap[def.esModule] === undefined) {
-        promises.push(System.import(def.esModule));
-      }
-    });
-
-    Promise.all(promises).then(modules => {
-      //widgets is an array of objects, see type below
-      const widgets: widgetType[] = [];
-
-      //Promise.all returns an array of resolved modules.
-      // Place into an object with key = module name to make it easier to access in the below widget loadinng loop
-      modules.map(mod => {
-        moduleMap[mod.name] = mod;
-      });
-
-      //config.widgets is an array of widget names
-      config.widgets.map(widgetName => {
-        //First see if name exists in coreWidgets
-        if (coreWidgets[widgetName]) {
-          widgets.push(coreWidgets[widgetName]);
-        } else {
-          const widget = externalWidgets[widgetName];
-          let Component: FunctionComponent =
-            moduleMap[widget.esModule].widgets[widget.name];
-          widget.component = () => <Component />;
-
-          widgets.push(widget);
-        }
-      });
-
-      setWidgets(widgets);
-    });
-  }, [config, setWidgets]);
+  const [tabHistory, setTabHistory] = React.useState({});
 
   function getInitialTab() {
-    if (config == undefined || widgets.length === 0) {
+    if (config == undefined || navbarItems.length === 0) {
       return 0;
+      //need to rename from widget to something else
     } else if (widgetPath == undefined) {
       return config.defaultTabIndex;
     } else {
-      return widgets.findIndex(element => element.path === "/" + widgetPath);
+      return navbarItems.findIndex(
+        element => element.path === "/" + widgetPath
+      );
     }
   }
 
-  const [tabHistory, setTabHistory] = React.useState({});
+  function getCoreView(name: string): ViewType {
+    let view: ViewType;
+    if (coreWidgets[name]) {
+      return coreWidgets[name];
+    }
+    if (coreDashboards[name]) {
+      view = {
+        ...coreDashboards[name],
+        component: () => <Dashboard dashboardConfig={coreDashboards[name]} />
+      };
+      return view;
+    }
+    return;
+    //TODO: if(coreMultiDashboards[view])
+  }
+
+  function getExternalView(config, name: string): ViewType {
+    let i = config.widgetDefinitions.findIndex(item => item.name === name);
+    let view: ViewType = { name: name, component: null };
+    if (i !== -1) {
+      view.component = () => (
+        <Widget widgetConfig={config.widgetDefinitions[i]} />
+      );
+    }
+    i = config.dashboardDefinitions.findIndex(item => item.name === view);
```

Comment:
hard to understand whats happening here, `item.name === view`, is this a bug?

Reply:
there was a problem there. it should have been item.name === name. 
Changed to the following:
`   i = config.dashboardDefinitions.findIndex(
      dashboardDefinition => dashboardDefinition.name === name
    );
`